Services/Services_feeds/lunarcrush_service_feed/lunarcrush_feed.py

```diff
@@ -0,0 +1,161 @@
+
+    def _initialize(self):
+        pass # Nothing to do
+
+    def _something_to_watch(self):
+        return bool(self.lunarcrush_coins)
+
+    def _get_sleep_time_before_next_wakeup(self):
+        return commons_enums.TimeFramesMinutes[self.refresh_time_frame] * commons_constants.MINUTE_TO_SECONDS
+
+    def get_data_cache(self, current_time: float, key: typing.Optional[str] = None):
+        if self.data_cache is None:
+            return None
+
+        if key is None:
+            return self.data_cache
+
+        coin, topic = key.split(";")
+        if topic == services_constants.LUNARCRUSH_COIN_METRICS and self.data_cache.get(services_constants.LUNARCRUSH_COIN_METRICS) is not None:
+            return [item for item in self.data_cache.get(services_constants.LUNARCRUSH_COIN_METRICS).get(coin) if item.time <= current_time]
+        return None
+
+    async def _get_coin_data(self, session: aiohttp.ClientSession, coin: str, start_date: datetime.datetime, end_date: datetime.datetime) -> bool:
+        self.logger.debug(f"Getting lunarcrush coin data for {coin} from {start_date.timestamp()} to {end_date.timestamp()}...")
+        api_url = f"https://lunarcrush.com/api3/public/coins/{coin}/time-series/v2?bucket=day&interval=1d&start={int(start_date.timestamp())}&end={int(end_date.timestamp())}"
+        async with session.get(api_url) as response:
+            if response.status != 200:
+                self.logger.error(f"Lunarcrush API request failed with status: {response.status}")
+                return False
+
+            coin_metrics_data = await response.json()
+            data = coin_metrics_data["data"]
+            self.data_cache[services_constants.LUNARCRUSH_COIN_METRICS] = {
+                coin: [
+                    LunarCrushCoinMetrics(
+                        time=entry["time"],
+                        contributors_active=entry["contributors_active"],
+                        contributors_created=entry["contributors_created"],
+                        interactions=entry["interactions"],
+                        posts_active=entry["posts_active"],
+                        posts_created=entry["posts_created"],
+                        sentiment=entry["sentiment"],
+                        spam=entry["spam"],
+                        alt_rank=entry["alt_rank"],
+                        circulating_supply=entry["circulating_supply"],
+                        close=entry["close"],
+                        galaxy_score=entry["galaxy_score"],
+                        high=entry["high"],
+                        low=entry["low"],
+                        market_cap=entry["market_cap"],
+                        market_dominance=entry["market_dominance"],
+                        open=entry["open"],
+                        social_dominance=entry["social_dominance"],
+                        volume_24h=entry["volume_24h"]
```

Comment:
if parsed object keys are == to dataclass keys, we could use https://github.com/Drakkar-Software/OctoBot-Commons/blob/master/octobot_commons/dataclasses/flexible_dataclass.py 
(make `LunarCrushCoinMetrics` inherit `FlexibleDataclass` and use `LunarCrushCoinMetrics.from_dict(entry)`).
The same is true for CoindeskNews